File: dongyimai-manager-web/src/main/java/com/senjie/sellergoods/controller/BrandController.java
package com.senjie.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.senjie.entity.PageResult;
import com.senjie.entity.Result;
import com.senjie.pojo.TbBrand;
import com.senjie.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * @Author SenJie
 * @Data 2021/3/30 16:13
 */
@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference(timeout = 30000)
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAllBrands () {
        return brandService.findAllBrands();
    }

    @RequestMapping("/findPage")
    public PageResult findPage (int page, int rows) {
        return brandService.findPage(page, rows);
    }

    @RequestMapping("/add")
    public Result add (@RequestBody TbBrand brand) {
        brandService.add(brand);

        try {
            return new Result(true, "  successfully added");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "   failure added");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne (Long id) {
        return brandService.findOne(id);
    }


    @RequestMapping("/update")
    public Result update (@RequestBody TbBrand brand) {
        brandService.update(brand);
        try {
            return new Result(true, "  successfully update");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "   failure update");
        }

    }

    @RequestMapping("/delete")
    public Result delete (Long[] id) {
        brandService.delete(id);
        try {
            return new Result(true, "  successfully delete");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "   failure delete");
        }
    }

    @RequestMapping("/search")
    public PageResult search (@RequestBody TbBrand brand, int page, int rows) {
        return brandService.findPage(brand, page, rows);

    }

    @RequestMapping("/findBrandList")
    public List<Map> findBrandList () {

        return brandService.findBrandList();
    }
}

File: dongyimai-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function ($scope,$controller ,brandService) {
    //继承
    $controller('baseController',{$scope:$scope});


    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;

            }
        )
    };

     //分页
    $scope.findPage = function (page, rows) {
        brandService.findPage(page, rows).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        )
    }

    $scope.save = function () {
        if ($scope.entity.id != null) {
            brandService.update($scope.entity).success(
                function (response) {
                    if (response.success) {
                        $scope.reloadList();
                        $scope.entity = {};
                    } else {
                        alert(response.message)
                    }
                })
        } else {

            brandService.add($scope.entity).success(
                function (response) {
                    if (response.success) {
                        $scope.reloadList();
                        $scope.entity = {};
                    } else {
                        alert(response.message)
                    }
                })
        }

    }
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;

            }
        )
    }

    $scope.delete = function () {
        brandService.delete($scope.selectId).success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                }

            })
    }
    $scope.serchEntity = {};
    $scope.search = function (page, rows) {
        brandService.search(page, rows, $scope.serchEntity).success(
            function (response) {
                $scope.paginationConf.totalItems = response.total;
                $scope.list = response.rows;
            }
        )
    }
});

